Firestore/Source/API/FIRQuery.mm

```diff
@@ -581,17 +600,19 @@ - (Bound)boundFromFieldValues:(NSArray<id> *)fieldValues isBefore:(BOOL)isBefore
                          "than were specified in the order by.");
   }
 
-  std::vector<FieldValue> components;
+  google_firestore_v1_ArrayValue components;
+  components.values_count = CheckedSize(fieldValues.count);
+  components.values = MakeArray<google_firestore_v1_Value>(components.values_count);
   for (NSUInteger idx = 0, max = fieldValues.count; idx < max; ++idx) {
     id rawValue = fieldValues[idx];
     const OrderBy &sortOrder = explicitSortOrders[idx];
 
-    FieldValue fieldValue = [self parsedQueryValue:rawValue];
+    google_firestore_v1_Value fieldValue = [self parsedQueryValue:rawValue];
     if (sortOrder.field().IsKeyFieldPath()) {
-      if (fieldValue.type() != FieldValue::Type::String) {
+      if (GetTypeOrder(fieldValue) != TypeOrder::kString) {
         ThrowInvalidArgument("Invalid query. Expected a string for the document ID.");
       }
-      const std::string &documentID = fieldValue.string_value();
+      std::string documentID = MakeString(fieldValue.string_value);
```

Comment:
If I recall correctly, you said that I am not able to use string_view APIs from iOS code. Is that correct?

Reply:
You cannot use Abseil in the calls between our code from `firebase-cpp-sdk` (aka public C++) and `firebase-ios-sdk` (aka core C++) because they are compiled separately. In this case, all the code would be within the iOS SDK and compiled together, so it's not an issue. However, making it possible for the public iOS types to access those APIs while hiding them from public C++ sounds like more trouble than it's worth.